src/material-examples/cdk-tree-flat/cdk-tree-flat-example.ts

```diff
@@ -117,33 +53,27 @@ export class FileDatabase {
   selector: 'cdk-tree-flat-example',
   templateUrl: 'cdk-tree-flat-example.html',
   styleUrls: ['cdk-tree-flat-example.css'],
-  providers: [FileDatabase]
 })
 export class CdkTreeFlatExample {
-  treeControl: FlatTreeControl<FileFlatNode>;
-  treeFlattener: MatTreeFlattener<FileNode, FileFlatNode>;
-  dataSource: MatTreeFlatDataSource<FileNode, FileFlatNode>;
-
-  constructor(database: FileDatabase) {
-    this.treeFlattener = new MatTreeFlattener(this.transformer, this._getLevel,
-      this._isExpandable, this._getChildren);
-    this.treeControl = new FlatTreeControl<FileFlatNode>(this._getLevel, this._isExpandable);
-    this.dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener);
-
-    database.dataChange.subscribe(data => {
-      this.dataSource.data = data;
-    });
+  private transformer = (node: FoodNode, level: number) => {
+    return {
+      expandable: !!node.children && node.children.length > 0,
+      name: node.name,
+      level: level,
+    };
   }
 
-  hasChild = (_: number, _nodeData: FileFlatNode) => _nodeData.expandable;
+  treeControl = new FlatTreeControl<ExampleFlatNode>(
+      node => node.level, node => node.expandable);
 
-  transformer = (node: FileNode, level: number) => {
-    return new FileFlatNode(!!node.children, node.filename, level, node.type);
-  }
+  treeFlattener = new MatTreeFlattener(
+      this.transformer, node => node.level, node => node.expandable, node => node.children);
 
-  private _getLevel = (node: FileFlatNode) => node.level;
+  dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener, TREE_DATA);
 
-  private _isExpandable = (node: FileFlatNode) => node.expandable;
+  constructor() {
+    this.dataSource.data = TREE_DATA;
```

Comment:
It seems like you're passing in the `TREE_DATA` in the initializer above already, it shouldn't be necessary to assign it again in here.

Reply:
you still need it, otherwise, it doesn't work. How about we do it this for now. I will look into this potential bug as I am doing some refactoring for tree.